File: jvm/core/main/scala/com/webank/eggroll/core/resourcemanager/job/JobProcessorTypes.scala
package com.webank.eggroll.core.resourcemanager.job

object JobProcessorTypes extends Enumeration {
  val DeepSpeed = Value

  def fromString(color: String): Option[JobProcessorTypes.Value] = {
    color.toLowerCase match {
      case "deepspeed" => Some(DeepSpeed)
      case _ => None
    }
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/JobProcessorTypes.java
package com.eggroll.core.pojo;


public enum JobProcessorTypes {
    DeepSpeed("deepspeed"),
    FlowJob("flowjob");

    private final String name;
    JobProcessorTypes(String name){
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static String fromString(String color) {
        switch (color.toLowerCase()) {
            case "deepspeed":
                return DeepSpeed.getName();
            default:
                throw new IllegalArgumentException("unsupported job processor type: " + color);
        }
    }
}
